application/src/Form/ResourceBatchUpdateForm.php

```diff
@@ -1,11 +1,329 @@
+                ]);
+                break;
+        }
+
+        $this->add([
+            'name' => 'clear_property_values',
+            'type' => PropertySelect::class,
+            'attributes' => [
+                'id' => 'remove-property-values',
+                'class' => 'chosen-select',
+                'multiple' => true,
+                'data-placeholder' => 'Select properties', // @translate
+            ],
+            'options' => [
+                'label' => 'Clear property values', // @translate
+            ],
+        ]);
+
+        // This hidden element manages the elements "value" added in the view.
+        $this->add([
+            'name' => 'value',
+            'type' => 'Hidden',
+            'attributes' => [
+                'value' => '',
+            ],
+        ]);
+
+        $addEvent = new Event('form.add_elements', $this);
+        $this->getEventManager()->triggerEvent($addEvent);
+
+        $inputFilter = $this->getInputFilter();
+        $inputFilter->add([
+            'name' => 'is_public',
+            'required' => false,
+        ]);
+        $inputFilter->add([
+            'name' => 'is_open',
+            'required' => false,
+        ]);
+        $inputFilter->add([
+            'name' => 'resource_template',
+            'required' => false,
+        ]);
+        $inputFilter->add([
+            'name' => 'resource_class',
+            'required' => false,
+        ]);
+        $inputFilter->add([
+            'name' => 'add_to_item_set',
+            'required' => false,
+        ]);
+        $inputFilter->add([
+            'name' => 'remove_from_item_set',
+            'required' => false,
+        ]);
+        $inputFilter->add([
+            'name' => 'clear_property_values',
+            'required' => false,
+        ]);
+        $inputFilter->add([
+            'name' => 'value',
+            'required' => false,
+        ]);
+
+        $filterEvent = new Event('form.add_input_filters', $this, ['inputFilter' => $inputFilter]);
+        $this->getEventManager()->triggerEvent($filterEvent);
+    }
+
+    /**
+     * @param Url $urlHelper
+     */
+    public function setUrlHelper(Url $urlHelper)
+    {
+        $this->urlHelper = $urlHelper;
+    }
+
+    /**
+     * @return Url
+     */
+    public function getUrlHelper()
+    {
+        return $this->urlHelper;
+    }
+
+    /**
+     * Preprocess data to get data to replace, to remove and to append.
+     *
+     * Batch update data contains instructions on what to update. It needs to be
+     * preprocessed before it's sent to the API. The elements are udpated by
+     * entity according to the attribute "data-collection-action", that can be
+     * "replace" (default), "remove" or "append".
+     *
+     * @todo Use standard validationGroup and filters.
+     * Note:
+     *
+     * @return array Associative array of data to replace, to remove and to
+     * append.
+     */
+    public function preprocessData()
```

Comment:
I agree with the logic behind moving this code from the disparate controllers to here. Perhaps it would be better suited as an input filter, but this is workable if that's too much trouble.

Reply:
This pr is associated with two other ones, that have the same aim and the same code: #1100 and #1133. Of course, any of my pr can be improved, by anybody.